flutter/lib/src/integrations/native_app_start_integration.dart

```diff
@@ -1,256 +1,79 @@
-
-  /// Flag indicating if app start was already fetched.
-  static bool _didFetchAppStart = false;
+  final NativeAppStartHandler _nativeAppStartHandler;
+  DateTime? _appStartEnd;
 
-  /// Flag indicating if app start measurement was added to the first transaction.
+  /// This timestamp marks the end of app startup. Either set by calling
+  // ignore: deprecated_member_use_from_same_package
+  /// [SentryFlutter.setAppStartEnd]. The [SentryFlutterOptions.autoAppStart]
+  /// option needs to be false.
   @internal
-  static bool didAddAppStartMeasurement = false;
-
-  /// Timeout duration to wait for the app start info to be fetched.
-  static const _timeoutDuration = Duration(seconds: 10);
-
-  @visibleForTesting
-  static Duration get timeoutDuration => _timeoutDuration;
-
-  /// We filter out App starts more than 60s
-  static const _maxAppStartMillis = 60000;
-
-  static Completer<AppStartInfo?> _appStartCompleter =
-      Completer<AppStartInfo?>();
-  static AppStartInfo? _appStartInfo;
-
-  @internal
-  static bool isIntegrationTest = false;
-
-  @internal
-  static void setAppStartInfo(AppStartInfo? appStartInfo) {
-    _appStartInfo = appStartInfo;
-    if (_appStartCompleter.isCompleted) {
-      _appStartCompleter = Completer<AppStartInfo?>();
+  set appStartEnd(DateTime appStartEnd) {
+    _appStartEnd = appStartEnd;
+    if (!_appStartEndCompleter.isCompleted) {
+      _appStartEndCompleter.complete();
     }
-    _appStartCompleter.complete(appStartInfo);
   }
 
-  @internal
-  static Future<AppStartInfo?> getAppStartInfo() {
-    if (_appStartInfo != null) {
-      return Future.value(_appStartInfo);
-    }
-    return _appStartCompleter.future
-        .timeout(_timeoutDuration, onTimeout: () => null);
-  }
-
-  @visibleForTesting
-  static void clearAppStartInfo() {
-    _appStartInfo = null;
-    _appStartCompleter = Completer<AppStartInfo?>();
-    didAddAppStartMeasurement = false;
-  }
-
-  /// Reset state
-  @visibleForTesting
-  static void reset() {
-    appStartEnd = null;
-    _didFetchAppStart = false;
-  }
+  final Completer<void> _appStartEndCompleter = Completer<void>();
 
   @override
-  void call(Hub hub, SentryFlutterOptions options) {
-    if (isIntegrationTest) {
-      final appStartInfo = AppStartInfo(
-        AppStartType.cold,
-        start: DateTime.now(),
-        end: DateTime.now().add(const Duration(milliseconds: 100)),
-        pluginRegistration:
-            DateTime.now().add(const Duration(milliseconds: 50)),
-        sentrySetupStart: DateTime.now().add(const Duration(milliseconds: 60)),
-        nativeSpanTimes: [],
-      );
-      setAppStartInfo(appStartInfo);
-      return;
-    }
-
-    if (_didFetchAppStart) {
-      return;
+  void call(Hub hub, SentryFlutterOptions options) async {
+    if (!options.autoAppStart && _appStartEnd == null) {
+      await callThrowing(options, () async {
+        await _appStartEndCompleter.future.timeout(const Duration(seconds: 10));
```

Comment:
can you check if this is still an issue: https://github.com/getsentry/sentry-dart/issues/2103

Reply:
had to move the completer into the post frame callback, otherwise we would block the ui again for the timeout duration, but it works now, also with #2103